src/t_set.c

```diff
@@ -256,7 +256,6 @@ int setTypeRemoveAux(robj *setobj, char *str, size_t len, int64_t llval, int str
     if (setobj->encoding == OBJ_ENCODING_HT) {
         sds sdsval = str_is_sds ? (sds)str : sdsnewlen(str, len);
         int deleted = (dictDelete(setobj->ptr, sdsval) == DICT_OK);
-        if (deleted && htNeedsResize(setobj->ptr)) dictResize(setobj->ptr);
```

Comment:
i'm paranoid about the reason why the approach we're taking here wasn't taken in the first place.

maybe you can compose a table indicating which place uses each approach?
i.e.

1. cron calls htNeedsResize
2. a call to htNeedsResize right after each dictDelete
3. dicts that use the above, but have some places that do dictDelete without doing htNeedsResize
4. dicts that are completely missing shrinking

maybe that will reveal something we're missing.